compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -857,9 +857,17 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                 }
 
                 // At this point, we are calling a function, `callee`, whose `DefId` is known...
-
                 if is_lang_panic_fn(tcx, callee) {
                     self.check_op(ops::Panic);
+
+                    // an internal `panic()` call must have exactly one argument
+                    // and evaluation of the `panic` op assumes the argument is `&str`
+                    match args[0].ty(&self.ccx.body.local_decls, tcx).kind() {
```

Comment:
I'm assuming this indexing is okay since the call would otherwise have been rejected by typechecking long before we got here.

Reply:
`is_lang_panic_fn` returns true for three different functions... does it really make sense to treat them all the same here? `panic_fn` and `panic_str` can never have any other argument I think, by mere type-checking. So the test probably should only fire for `begin_panic_fn`.